File: drift/src/main/java/drift/com/drift/helpers/DateHelper.java
package drift.com.drift.helpers;

import android.text.format.DateFormat;

import java.text.SimpleDateFormat;
import java.util.Date;

import drift.com.drift.Drift;



public class DateHelper {
    public static boolean isSameDay(Date dateOne, Date dateTwo) {

        SimpleDateFormat fmt = new SimpleDateFormat("yyyyMMdd");

        if (fmt.format(dateOne).equals(fmt.format(dateTwo))) {
            return true;
        }
        return false;
    }

    public static String formatDateForConversation(Date date){

        if (isSameDay(date, new Date())){
            return DateFormat.getTimeFormat(Drift.getContext()).format(date);
        }else{
            return DateFormat.getDateFormat(Drift.getContext()).format(date);
        }
    }

    public static String formatDateForScheduleDay(Date date){
        return DateFormat.getDateFormat(Drift.getContext()).format(date);
    }

    public static String formatDateForScheduleTime(Date date){
        return DateFormat.getTimeFormat(Drift.getContext()).format(date);
    }
}

File: drift/src/main/java/drift/com/drift/helpers/DateHelper.kt
package drift.com.drift.helpers

import android.content.Context
import android.text.format.DateFormat

import java.text.SimpleDateFormat
import java.util.Date

import drift.com.drift.Drift


internal object DateHelper {
    fun isSameDay(dateOne: Date?, dateTwo: Date?): Boolean {

        if (dateOne == null || dateTwo == null) {
            return false
        }

        val fmt = SimpleDateFormat("yyyyMMdd")

        return fmt.format(dateOne) == fmt.format(dateTwo)
    }

    fun formatDateForConversation(context: Context, date: Date?): String {

        if (date == null){
            return ""
        }

        return if (isSameDay(date, Date())) {
            DateFormat.getTimeFormat(context).format(date)
        } else {
            DateFormat.getDateFormat(context).format(date)
        }
    }

    fun formatDateForScheduleDay(context: Context, date: Date): String {
        return DateFormat.getDateFormat(context).format(date)
    }

    fun formatDateForScheduleTime(context: Context, date: Date): String {
        return DateFormat.getTimeFormat(context).format(date)
    }
}